src/openrct2/rct2/S6Importer.cpp

```diff
@@ -296,7 +296,12 @@ class S6Importer final : public IParkImporter
         gScenarioObjectiveYear = _s6.objective_year;
         // pad_013580FA
         gScenarioObjectiveCurrency = _s6.objective_currency;
-        gScenarioObjectiveNumGuests = _s6.objective_guests;
+        // In RCT2, the ride string IDs start at index STR_0002 and are directly mappable.
+        // This is not always the case in OpenRCT2, so we use the actual ride ID.
+        if (gScenarioObjectiveType == OBJECTIVE_BUILD_THE_BEST)
+            gScenarioObjectiveNumGuests = _s6.objective_guests - RCT2_RIDE_STRING_START;
```

Comment:
If you are going to use this for not just guests we should rename this so that it is just a objective argument instead of num guests.

Reply:
It is also used for the length requirement for roller coasters.

I have a pull request coming up that creates a struct for the objective, in which I also handle this.

Reply:
Okay that's fine then